Securing Sensu Go (#926)
sensu/sensu-docs: content/sensu-go/5.0/guides/securing-sensu-go.md

---

> title: "How to Secure Sensu Go"

Reviewer: ```suggestion
title: "Securing Sensu"
```

> linkTitle: "Securing Sensu Go"

Reviewer: ```suggestion
```

weight: 12
version: "5.0"
product: "Sensu Go"
platformContent: false
menu:
  sensu-go-5.0:
    parent: guides
---

> As with any piece of software, it is critical to minimize any attack surface exposed by the software. Sensu is no different. If you are familiar with Sensu, you are already aware of the [component pieces that need to be secured][1]. Sensu Go, while different its deployment architecture, still contains component pieces that will need to be secured in order to be considered "production ready":

Reviewer: ```suggestion
As with any piece of software, it is critical to minimize any attack surface exposed by the software. Sensu is no different. The following component pieces need to be secured in order for Sensu to be considered production ready:
```

* Sensu agent to server communication
* etcd peer communication
* Backend API
* Dashboard

We'll cover securing each one of those pieces, starting with Sensu agent to server communication.

## Securing Sensu agent to server communication

> The Sensu agent uses WebSockets for communication between the agent and the server. By default, an agent will use the insecure `ws://` transport:

Reviewer: ```suggestion
The Sensu agent uses WebSockets for communication between the agent and the server. By default, an agent uses the insecure `ws://` transport:
```

{{< highlight shell >}}
---
##
# agent configuration
##
...
backend-url:
  - "ws://127.0.0.1:8081"

> {{< /highlight >}}

Reviewer: This sample needs some context: that it's the agent config located at...

In order to use WebSockets over SSL/TLS (wss), change the `backend-url` value to the `wss://` schema:

{{< highlight shell >}}
---
##
# agent configuration
##
...
backend-url:
  - "wss://127.0.0.1:8081"
{{< /highlight >}}

The agent will then connect Sensu servers over wss. Let's move on to securing etcd peer communication.

> _NOTE: If creating a Sensu Go cluster, every cluster member will need to be present in the configuration. See the [Sensu Go clustering guide][2] for more information on how to configure agents for a clustered configuration._

Reviewer: ```suggestion
_NOTE: If creating a Sensu cluster, every cluster member needs to be present in the configuration. See the [Sensu Go clustering guide][2] for more information on how to configure agents for a clustered configuration._
```
